src/plugins/data_views/common/data_views/data_view.ts

```diff
@@ -132,6 +132,7 @@ export class DataView implements DataViewBase {
    * Prevents errors when index pattern exists before indices
    */
   public readonly allowNoIndex: boolean = false;
+  public name: string = '';
```

Comment:
Could you add inline docs? It would be nice not to backtrack on the effort to add docs.